In [1]:
import csv
import pandas as pd
import glob
import html
import re
from tqdm import tqdm_notebook as tqdm

### Made the full merged_christmas file through this piece of code (don't run it constantly, otherwise it just adds to the csv file, if necessary delete the csv file and rerun this code).

In [ ]:
#fout=open("merged_christmas.csv","a")
# first file:
#for line in open("christmas_datasets/reviews_A_Christmas_Carol__Colorized___VHS_.csv"):
    #fout.write(line)
# now the rest: 

#other_movies = ['A_Christmas_Story__VHS_', 'Miracle_on_34th_Street__VHS_', 'National_Lampoon_s_Christmas_Vacation__VHS_', 'The_Chronicles_of_Narnia_-_The_Lion__the_Witch_and_the_Wardrobe', 'Tim_Burton_s_The_Nightmare_Before_Christmas__VHS_', 'White_Christmas__VHS_']
#for i in other_movies:
    #f = open("christmas_datasets/reviews_"+(i)+".csv")
    #next(f) # skip the header
    #for line in f:
        
         #fout.write(line)
    #f.close() # not really needed

#fout.close()

### now off to alter the review text (html.unescape) and add it to a dataframe

In [2]:
df = pd.read_csv('other_datasets/others.csv', sep='\t', converters={'review_text':html.unescape, 'reviewer_name':html.unescape, 'summary':html.unescape})

df=df[~df['rating'].isin(['rating'])]
#df=df[~df['rating'].isin(['1','2','3','4','5','rating'])]
df

asin                        title     reviewer_id  \
0       6302760046  2001: A Space Odyssey [VHS]  A319PM1MXFDS03   
1       6302760046  2001: A Space Odyssey [VHS]  A1ALOIPRR8Q06J   
2       6302760046  2001: A Space Odyssey [VHS]  A1Y09QLADQYQJG   
3       6302760046  2001: A Space Odyssey [VHS]  A36ZLE3M589MFL   
4       6302760046  2001: A Space Odyssey [VHS]  A1NOLEWY3EBW01   
5       6302760046  2001: A Space Odyssey [VHS]  A3TS93NJJ7AADI   
6       6302760046  2001: A Space Odyssey [VHS]  A24QKJEGSWTGPE   
7       6302760046  2001: A Space Odyssey [VHS]   AAKF9UNZU7AFO   
8       6302760046  2001: A Space Odyssey [VHS]  A1AMETU2F0W5DQ   
9       6302760046  2001: A Space Odyssey [VHS]  A1KEYYYPNKNM65   
10      6302760046  2001: A Space Odyssey [VHS]  A3HQXFLZ6WI70Q   
11      6302760046  2001: A Space Odyssey [VHS]  A3BZTIB28MC82V   
12      6302760046  2001: A Space Odyssey [VHS]  A32JKNQ6BABMQ2   
13      6302760046  2001: A Space Odyssey [VHS]  A1QHHSQ3YIDED1   
14      6302760046  2001: A Space Odyssey [VHS]   A3OZES3R70ZMQ   
15      6302760046  2001: A Space Odyssey [VHS]  A26Z184WOGF8CM   
16      6302760046  2001: A Space Odyssey [VHS]  A2PS1KAU6MGELR   
17      6302760046  2001: A Space Odyssey [VHS]  A2JJOLN5OIUCOI   
18      6302760046  2001: A Space Odyssey [VHS]   A32T89D22H678   
19      6302760046  2001: A Space Odyssey [VHS]  A1G3TE56DN52ME   
20      6302760046  2001: A Space Odyssey [VHS]  A2B09YP4YMLSSU   
21      6302760046  2001: A Space Odyssey [VHS]  A2ZYLTXBHUFVBH   
22      6302760046  2001: A Space Odyssey [VHS]  A3PUA1HHTUULM7   
23      6302760046  2001: A Space Odyssey [VHS]  A2PV6GK1HV54Y9   
24      6302760046  2001: A Space Odyssey [VHS]  A3MODJJA6RLGJ3   
25      6302760046  2001: A Space Odyssey [VHS]  A3VSXJ959S2HY0   
26      6302760046  2001: A Space Odyssey [VHS]  A1GSR7RGCG1QYZ   
27      6302760046  2001: A Space Odyssey [VHS]  A1LKI2DJVA5ATE   
28      6302760046  2001: A Space Odyssey [VHS]  A2FYML6MFOUXJ1   
29      6302760046  2001: A Space Odyssey [VHS]  A27FC75N4V9CKZ   
...            ...                          ...             ...   
235254  6305368139        You've Got Mail [VHS]  A1YYYR4OVR3E7J   
235255  6305368139        You've Got Mail [VHS]  A33WAM61OVNOUG   
235256  6305368139        You've Got Mail [VHS]    AI4PK0YTFJ3G   
235257  6305368139        You've Got Mail [VHS]  A34ZGYGL441YQD   
235258  6305368139        You've Got Mail [VHS]  A1F12E9U638Y66   
235259  6305368139        You've Got Mail [VHS]  A3TUJ8M4WUHVT4   
235260  6305368139        You've Got Mail [VHS]  A311HPT8X8XFF7   
235261  6305368139        You've Got Mail [VHS]  A11DESP9TFR44R   
235262  6305368139        You've Got Mail [VHS]  A1WG3XQA6L7TV0   
235263  6305368139        You've Got Mail [VHS]  A18N6AU5O4WX6Q   
235264  6305368139        You've Got Mail [VHS]  A2A5LQASP42HND   
235265  6305368139        You've Got Mail [VHS]   A6P3IHM0T0A2Q   
235266  6305368139        You've Got Mail [VHS]  A217C5LB9ZE9OF   
235267  6305368139        You've Got Mail [VHS]  A22IV5UBMWEKWE   
235268  6305368139        You've Got Mail [VHS]  A1CM2MXGH3SV19   
235269  6305368139        You've Got Mail [VHS]  A1I67DTCROOS25   
235270  6305368139        You've Got Mail [VHS]  A1S93M3BYF5Y6E   
235271  6305368139        You've Got Mail [VHS]  A1CAF866HYZTHA   
235272  6305368139        You've Got Mail [VHS]   AKF278ZC318PS   
235273  6305368139        You've Got Mail [VHS]  A38H6JJZM0R2V1   
235274  6305368139        You've Got Mail [VHS]   ALV2TIVYOYF3O   
235275  6305368139        You've Got Mail [VHS]  A2N86KJSL3JBG9   
235276  6305368139        You've Got Mail [VHS]   ABYAPGN7LXZGZ   
235277  6305368139        You've Got Mail [VHS]  A3PIA8Z2DNNILT   
235278  6305368139        You've Got Mail [VHS]  A3Q8QAG0OSGMT6   
235279  6305368139        You've Got Mail [VHS]  A240JSQJGOTGMD   
235280  6305368139        You've Got Mail [VHS]  A3BVEEN4N6B11R   
235281  6305368139        You've Got Mail [VHS]   AFZNJHDGE0D

### let's do some calculations

In [5]:
import numpy as np
from scipy.stats import f_oneway

actually_december = []
not_december = []

for index, row in tqdm(df.iterrows()):
    
    if re.match('[0-9]+-12-(2[4-9]|30)', row['review_date']):
        actually_december.append(float(row['rating']))
    else:
        not_december.append(float(row['rating']))
            
print("actually_december:", round(sum(actually_december)/float(len(actually_december)), 2)," vs not_december:", round(sum(not_december) / float(len(not_december)), 2))
print("number of ratings:", len(actually_december),len(not_december))

print(f_oneway(np.array(actually_december),np.array(not_december)))

HBox(children=(IntProgress(value=1, bar_style='info', max=1), HTML(value='')))


actually_december: 4.37  vs not_december: 4.36
number of ratings: 6355 228788
F_onewayResult(statistic=0.55247887357200631, pvalue=0.45730733574214155)


## Christmas time = 12-24 to 12-30
### Christmas movies:

actually_december: 4.58 vs not_december: 4.65  
number of ratings: 1283 8566  
F_onewayResult(statistic=6.6695623219982476, pvalue=0.009821624109377685)  

### Non-Christmas movies:

actually_december: 4.37 vs not_december: 4.36  
number of ratings: 6355 228788  
F_onewayResult(statistic=0.55247887357200631, pvalue=0.45730733574214155)  
